src/main/java/io/vertx/core/http/impl/ConnectionManager.java

```diff
@@ -53,6 +51,18 @@
     this.version = version;
   }
 
+  synchronized void start(boolean checkExpired) {
+    this.timerID = checkExpired ? client.getVertx().setTimer(1, id -> checkExpired()) : -1;
+  }
+
+  private synchronized void checkExpired() {
+    if (timerID >= 0) {
```

Comment:
Would you please elaborate on the use of this checkExpired()? what exactly is expired, and why Is it synchronized, is this a stop whole app synchronized. (If It's the timeID, the you may consider AtomicInteger ?). What do you think?

Reply:
it checks the `HttpClientConnection` which have been in the pools for too long so these connections are closed

synchronized is actually better because we are used biased locking and always the same thread, it's fast because not contended

Reply:
it is to evict unused keep-alive connections that have been idle for a long time

so actually, it may be called from any thread, but we expect it will be always called from the same thread when it is used inside a verticle